### PR TITLE
TRM-27170: commonTaxon UniRef API light/non light consistency

### PR DESCRIPTION
**Pull request by LeonardoGonzales**

### CODE REVIEW

File: xml-parser/src/test/java/org/uniprot/core/xml/uniref/UniRefEntryConverterTest.java [hunk 1]
```diff
@@ -41,6 +48,69 @@ void testFromXml() throws Exception {
...
+                                        .organismTaxId(9606L)
+                                        .build())
+                        .build();
+        Entry xmlConverter = converter.toXml(uniRefEntry);
+        assertNotNull(xmlConverter);
+        UniRefEntry entryConverted = converter.fromXml(xmlConverter);
+        assertNotNull(entryConverted);
+        assertEquals(uniRefEntry, entryConverted);
+    }
+
```
Comment (ahmadshadab) [inline xml-parser/src/test/java/org/uniprot/core/xml/uniref/UniRefEntryConverterTest.java:113]: Is it possible to have commanTaxon  without either taxonId  or  scientificName  or  both? If yes, we should cover that in our  test coverage.

File: core-domain/src/test/java/org/uniprot/core/uniref/impl/UniRefEntryBuilderTest.java [hunk 2]
```diff
@@ -78,20 +83,13 @@ void testUpdated() {
...
     void testCommonTaxonName() {
-        String commonTax = "Homo sapiens";
+        Organism organism =
+                new OrganismBuilder().taxonId(9606L).scientificName("scientific").build();
 
-        UniRefEntry entry = new UniRefEntryBuilder().commonTaxon(commonTax).build();
-        assertEquals(commonTax, entry.getCommonTaxon());
+        UniRefEntry entry = new UniRefEntryBuilder().commonTaxon(organism).build();
+        assertEquals(organism, entry.getCommonTaxon());
     }
```
Comment (ahmadshadab) [inline core-domain/src/test/java/org/uniprot/core/uniref/impl/UniRefEntryBuilderTest.java:94]: Could we add test case without either taxonId or scientificName or both?

File: xml-parser/src/test/java/org/uniprot/core/xml/uniref/UniRefEntryConverterTest.java (hunk 1 above)
Comment (ahmadshadab) [reply xml-parser/src/test/java/org/uniprot/core/xml/uniref/UniRefEntryConverterTest.java:113]: Leo one more thing, Aurelien has mentioned about making uniparcId as String field. I didn't see changes related to  that.  Are we not doing that?

Comment (LeonardoGonzales) [reply xml-parser/src/test/java/org/uniprot/core/xml/uniref/UniRefEntryConverterTest.java:113]: The change for uniparcId  is in the class: UniRefEntryLightJsonConfig.java,  line 98

File: core-domain/src/test/java/org/uniprot/core/uniref/impl/UniRefEntryBuilderTest.java (hunk 2 above)
Comment (LeonardoGonzales) [reply core-domain/src/test/java/org/uniprot/core/uniref/impl/UniRefEntryBuilderTest.java:94]: This use case is covered in OrganismBuilderTest, right?

Comment (ahmadshadab) [reply core-domain/src/test/java/org/uniprot/core/uniref/impl/UniRefEntryBuilderTest.java:94]: Oh sorry, i didnt see that. If that case is covered please ignore the comment

File: xml-parser/src/test/java/org/uniprot/core/xml/uniref/UniRefEntryConverterTest.java (hunk 1 above)
Comment (ahmadshadab) [reply xml-parser/src/test/java/org/uniprot/core/xml/uniref/UniRefEntryConverterTest.java:113]: All good Leo. Please merge.

**Pull request merged**